learning.py

```diff
@@ -0,0 +1,81 @@
+import nn_architecture
+import data_loader as dl
+
+import logging
+import numpy as np
+import copy
+
+import torch
+from torch import nn
+import torch.optim as optim
+from torch.optim import lr_scheduler
+from torch.utils.data import Dataset, DataLoader
+
+
+def train(model, x, y, matrix):
```

Comment:
Still plan to pass to the function a few extra variables:
`num_epochs`, `optimizer`, `scheduler`, `criterion`?

Reply:
vars are now passed to the Learner class